kopf/clients/patching.py

```diff
@@ -31,5 +34,14 @@
     cls = classes._make_cls(resource=resource)
     obj = cls(api, body)
 
-    loop = asyncio.get_running_loop()
-    await loop.run_in_executor(config.WorkersConfig.get_syn_executor(), obj.patch, patch)
+    # The handler could delete its own object, so we have nothing to patch. It is okay, ignore.
+    try:
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(config.WorkersConfig.get_syn_executor(), obj.patch, patch)
+    except pykube.ObjectDoesNotExist:
+        pass
+    except requests.exceptions.HTTPError as e:
```

Comment:
can you explain in what circumstances this exception might raise?

Reply:
Indeed! Good that you've noticed that.

Pykube has [its own handling of _all_ exceptions](https://github.com/hjacobs/pykube/blob/2ff99db72be0207af4f0c6e2c6d82adbeeb63559/pykube/http.py#L242-L249), and overrides it with its own `pykube.HTTPError` _only_ if they are applicaton/json and its json has `kind=="Status"` — regardless of the status code. I don't know how reliable this condition is, so it might happen that the `requests`'s exceptions leak out.

But that pykube's exception should be caught too, same as in the `auth.py` (on verification).

And, unlike the GET-methods in `fetching.py`, `pykube.ObjectDoesNotExist` will not be raised in PATCH/POST/PUT-methods. But that might change in the future, so we at least expect it (for forward-compatibility).

Reply:
Fixed with a new commit.

Reply:
Right, being more conservative in there regards shouldn't hurt